### DISCUSSION TITLE
Brownie:  command not found

### DISCUSSION THREAD

**Discussion by ChristianG77** [smartcontractkit/full-blockchain-solidity-course-py#336] [Q&A] [answered]
Hi,

VS code using WSL was working fine a few hours ago, and now I'm getting a 'brownie:  command not found' message in my terminal.  The only change I've made is updating Express VPN which I've now uninstalled, but still get the same message.

I also tried to update ubunto and the additional steps in the link below and received a decent amount of errors.

https://medium.com/@cromewar/how-to-setup-windows-10-11-for-smart-contract-development-and-brownie-e7d8d13555b3

Anyone having similar issues?

![image](https://user-images.githubusercontent.com/91568178/141027481-14c3f776-d47a-4d3f-9740-0aac4089e962.png)

![image](https://user-images.githubusercontent.com/91568178/141027649-91dd2317-4050-483b-9535-a5db28471b71.png)

**Comment by cromewar** [comment, marked as answer]
The ubuntu problem may be because the mirror list are pointing somewhere else (the location your vpn was pointing)
restart your computer and try again with update and upgrade.
Regarding to brownie, well it's estrange did you use a virtualenv? anyways just re-install it with `pip install eth-brownie`.

**Comment by ChristianG77** [comment]
Thanks and restarted computer as suggested.

Ya it's strange 'sudo apt upgrade' worked but errors on 'sudo apt-get update'.

And with 'pip install eth-brownie' is saying 'Requirement already satisfied'.

Usually when you type 'brownie' into the terminal, it highlights yellow, but not happening now.

![image](https://user-images.githubusercontent.com/91568178/141036308-0354c162-9b9b-4cd8-909a-aa02cca5c96a.png)

![image](https://user-images.githubusercontent.com/91568178/141036572-531af305-0d51-42fd-865b-1b54792c780a.png)

**Comment by cromewar** [comment]
Don't forget tu run the "fix missing" also as admin with sudo. If it says
is already satisfied could try to run brownie networks list?

**Reply by ChristianG77** [reply]
Thanks for the suggestions.  

I've tried them all and no luck.  

Maybe uninstall WSL, ubuntu, VS Code, Python and Brownie and reinstall them all again?

**Comment by cromewar** [comment]
It is not a ideal scenario, but yes.. give a try

**Reply by ChristianG77** [reply]
Ok cool I'll try that.

Should I remove everything with regards to VS Code including things stored in AppData/Local and AppData/Roaming including npm, pip, Yarn etc. that don't get automatically deleted with WSL, ubuntu, VS Code, Python etc?

Example as per screen shot below.

![image](https://user-images.githubusercontent.com/91568178/141210507-dcdb7a67-b36a-4614-afde-4b676c8d1c94.png)

**Reply by ChristianG77** [reply]
Should I also delete Ganache and Node.js?

**Reply by cromewar** [reply]
That's up to you, remember there are two different operating systems so if your not going to use those tools anymore on windows you can delete them, WSL will use their owns anyways.

**Reply by ChristianG77** [reply]
Thanks @cromewar for continuing to help me out with all my questions.

Got my comp back up and running after the reinstall!!  I think part of the problem was the path wasn't correct on the environment variables.  

I've run a test on Lesson 12's deploy_and_upgrade.py.  It works when I deploy to rinkeby but an error on Ganache saying 'ValueError:  Explorer API not set for this network.'

![image](https://user-images.githubusercontent.com/91568178/141267381-d1416693-64c3-4e22-90d1-b29b0ee4bfa4.png)

**Reply by cromewar** [reply]
Did you install ganache-cli using npm? could you please run just ganache-cli and see if it works properly?
also you might have residual files from brownie configs so delete them with:

`rm -rf ~/.brownie`

and try again.

**Reply by ChristianG77** [reply]
Ok so I tried ganache-cli and got the first screen shot below.

I also ran rm-rf ~/.brownie.

I ran the deploy with ganache and noticed before the error that the --port in the 'Lauching' line says 8545 instead of 7545 as I specified in my 'brownie networks modify development host=http://172.30.144.1:7545'.  Could the --port 8545 be the issue?

![image](https://user-images.githubusercontent.com/91568178/141378042-da1d8a0d-019d-4186-991f-5cd380bab8c7.png)

![image](https://user-images.githubusercontent.com/91568178/141378118-16aa0064-8064-4f85-b490-9eab3ac61617.png)

**Reply by cromewar** [reply]
100% yes, if you don't specify the network with `--network` brownie is going to select the default one and if there is a mismatch with the ports you are going to have troubles, also is a bad idea to modify the development network, you should create new ones... anyways try to match the port numbers.

**Comment by ChristianG77** [comment]
Ok thanks.

I was modifying the development network to meet the WSL hostname in ganache.

Is there a better way to specify this hostname and port number?  

![image](https://user-images.githubusercontent.com/91568178/141379132-daf3b5a6-a613-4403-afdb-3bdd78c9ab4c.png)

**Reply by cromewar** [reply]
Well, your best option is to use different networks, most of the time you will use ganache-cli and not the user interface, so is better just to add more networks to brownie with `brownie networks add`, you would like to have one for ganache-ui with wsl, another one (the defualt) for ganache-cli and any other ones you would need.

**Reply by ChristianG77** [reply]
I tried the 'brownie networks add' but I'm getting something wrong on the inputs.

I was able to get 'deploy_box.py' working using 'brownie networks modify development host=http://172.29.240.1:7545 port=7545'.  If 'modify' is not good, what should I be inputting after 'brownie networks add'?

I then tried running 'deploy_box_and_upgrade.py' but received 'ValueError: Explorer API not set for this network'.  The error is probably something to do with ganache as it works fine on rinkeby.
 
![image](https://user-images.githubusercontent.com/91568178/141423030-2548a458-c330-4624-aee0-8e564f9ef1a4.png)

**Reply by cromewar** [reply]
you should reinstall ganache:

`npm uninstall ganache-cli` and re install it, as you said is something wrong with your ganache.

btw: using modify is ok, also you can refer to [this](https://eth-brownie.readthedocs.io/en/stable/network-management.html)

**Reply by ChristianG77** [reply]
HI @cromewar.  

I uninstalled and reinstalled ganache, ran deploy_box.py successfully but still get 'ValueError:  Explorer API not set for this network' when I run the script for deploy_and_upgrade.py in lesson 12.

Any ideas?

![image](https://user-images.githubusercontent.com/91568178/142827281-e66160a0-e973-42d1-8ad2-adfe6f170651.png)

**Reply by cromewar** [reply]
remember to export de env variables on wsl with export variablename=value.
